geode-junit/src/main/java/org/apache/geode/internal/AvailablePortHelper.java

```diff
@@ -147,16 +139,15 @@ private static void releaseKeepers(List<Keeper> keepers) {
       site = Integer.parseInt(hostName.substring(4));
     }
 
-    Set set = new HashSet();
-    while (set.size() < count) {
-      int port = AvailablePort.getRandomAvailablePortWithMod(AvailablePort.SOCKET, site);
-      set.add(new Integer(port));
-    }
-    int[] ports = new int[set.size()];
+    int[] ports = new int[count];
     int i = 0;
-    for (Iterator iter = set.iterator(); iter.hasNext();) {
-      ports[i] = ((Integer) iter.next()).intValue();
-      i++;
+    while (i < count) {
+      int port = getUniquePort(false, AvailablePort.SOCKET);
+      while (port % site != 0) {
```

Comment:
This `site` logic looks wrong, and I'm pretty sure that it's legacy from the pre-Geode days. Can we remove it?

Reply:
Yeah, this is definitely wrong, but it's not clear what the right behavior is (well, it should be port % FOO == site, but it's unclear what FOO is).  The bigger problem is that this is long standing behavior (logic copied from AvailablePort.getRandomAvailablePortWithMod) that's called in over 60 places from over a dozen tests.  While the intended protection should be redundant with the new behavior I'm introducing, I think there may be older pieces of code using this that aren't going to be calling initializeUniquePortRange.  I'm not sure whether any of them depend on the current behavior and would become flakier if this was changed to just use a random port.

I would like to change this, but just have no confidence that doing so isn't going to introduce some unintended flakiness somewhere in the test code (particularly worried about hydra).